File: Stack/Stack.js
// create class with constructor for properties: items/data and size/count/top of array
class Stack {
  constructor() {
    this.items = [];
    this.size = 0;
  }

  // create methods for push, pop, and peek
  // when pushing increment the count/size
  push(element) {
    this.items[this.size] = element;
    this.size++;
    console.log(`added element: ${element}`);
  }

  // when popping/deleting decrement the count/size
  // delete item
  pop() {
    this.size--;
    console.log(`removed element: ${this.items.pop()}`);
    return this.items.pop();
  }

  peek() {
    console.log(`peek element: ${this.items[this.items.length - 1]}`);
    return this.items[this.items.length - 1];
  }

  // if the size === 0 then stack is empty
  isEmpty() {
    return this.size === 0;
  }
}

// test stack by creating new one
const stack = new Stack();

// perform operations
stack.isEmpty();

stack.push("dog");
stack.push("cat");

stack.peek()

stack.push("bear");
stack.push("owl")

stack.pop("cat");

File: Stack/Stack.c
#include <stdio.h>

int MAX_SIZE = 8;
int stack[8];
// when top = -1, stack is empty
// top is same as size in the js file
int top = -1;

// return element at the top of stack without modifying
int peek()
{
    return stack[top];
}

// cannot add if stack is full
// not full then push element
int push(int element)
{
    // top < size - 1 makes sure stack isn't full
    if (top < MAX_SIZE - 1)
    {
        top++;
        stack[top] = element;
        printf("added element: %d \n", element);
    }

    else
    {
        printf("Stack Overflow!!! \n");
    }
}

// cannot remove if empty
// not empty then pop element
int pop(int element)
{

    if (top >= 0)
    {
        stack[top] = element;
        top--;
        printf("removed element: %d \n", element);
    }
    else
        printf("Stack is empty!!!");
}

// test stack by performing operations
int main()
{
    printf("Top of Stack is: %d\n", peek());

    push(3);
    push(6);
    push(9);
    push(11);

    pop(9);

    return 0;
}